sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -1798,7 +1807,7 @@ class SessionCatalog(
     functions.map {
       case f if FunctionRegistry.functionSet.contains(f) => (f, "SYSTEM")
       case f if TableFunctionRegistry.functionSet.contains(f) => (f, "SYSTEM")
-      case f => (f, "USER")
+      case f => (attachSessionCatalog(f), "USER")
```

Comment:
another case we need to take care is temp function, which doesn't belong to any catalog.

It's probably fine here, as temp function identifier does not have database field, so attachSessionCatalog is a noop.